app/models/spree/variant_decorator.rb

```diff
@@ -99,19 +98,6 @@ def fees_by_type_for(distributor, order_cycle)
     OpenFoodNetwork::EnterpriseFeeCalculator.new(distributor, order_cycle).fees_by_type_for self
   end
 
-  def delete
-    if product.variants == [self] # Only variant left on product
-      errors.add :product, I18n.t(:spree_variant_product_error)
-      false
-    else
-      transaction do
-        self.update_column(:deleted_at, Time.zone.now)
-        ExchangeVariant.where(variant_id: self).destroy_all
```

Comment:
what happens with these exchange variants then? Are they destroyed on cascade somewhere else?

Reply:
Yes, they are destroyed as dependents. I added a spec for that.